drivers/lora/sx126x.c

```diff
@@ -358,6 +371,60 @@ static int sx126x_irq_init(void)
 	return 0;
 }
 
+#elif defined(SX126X_STM32WL)
```

Comment:
I would suggest moving these to a separate C file (`sx126x_standalone.c` and `sx126x_stm32wl.c`) and let the build system select the right implementation. That avoids clutter in this file and scales better in case there are more SoCs with similar requirements.

Reply:
I generally share the feeling, however at this stage, with only one compact alternative implementation I'd rather keep the whole driver it in a single file, and split it once and if a more special versions with different quirks shows up on the market. Unless you are aware of that happening already...

Reply:
I think it makes sense to do it now itself as it makes the driver look cleaner.

Reply:
Alright, moved few things around, see my notes in the comment after the push.